code/script.js

```diff
@@ -269,57 +286,90 @@ const filterCharacters = (keep) => {
   if (category === 'accessories') {
     if (keep) {
       alert(
-        `Yes, the person wears ${value}! Keep all people that wears ${value}`
+        `Yes, the person wears ${value}! Keep all people that wears ${value}!`
       )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
       alert(
-        `No, the person doesn't wear ${value}! Remove all people that wears ${value}`
+        `No, the person doesn't wear ${value}! Remove all people that wears ${value}!`
       )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
     }
   } else if (category === 'other') {
-    // Similar to the one above
-  } else {
     if (keep) {
-      // alert popup that says something like: "Yes, the person has yellow hair! Keep all people with yellow hair"
+      alert(
+        `Yes, that person is ${value}. Byebye all people who aren't ${value}!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category].includes(value))
     } else {
-      // alert popup that says something like: "No, the person doesnt have yellow hair! Remove all people with yellow hair"
+      alert(
+        `Noooope, ${value} is not correct! Byebye all people who are ${value}!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => !person[category].includes(value))
+    }
+  } else if (category === 'hair') {
+    if (keep) {
+      alert(
+        `How did you know!?! That person has ${value} hair. Lets keep 'em ${value}haired!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `No, wrong! ${value} is not correct, toss all the ${value}haired people!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
+    }
+  } else if (category === 'eyes') {
+    if (keep) {
+    alert(
+      `Yes, our secret person does have ${value} eyes. So long to all the other ones!`
+      )
+    charactersInPlay = charactersInPlay.filter((person) => person[category] === value)
+    } else {
+      alert(
+        `Wrong! Our secret person does not have ${value} eyes. Say godbye to all people with ${value} eyes!`
+        )
+      charactersInPlay = charactersInPlay.filter((person) => person[category] !== value)
     }
   }
 
-  // Determine what is the category
-  // filter by category to keep or remove based on the keep variable.
-  /* 
-    for hair and eyes :
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] === value)
-      or
-      charactersInPlay = charactersInPlay.filter((person) => person[attribute] !== value)
+  generateBoard()
```

Comment:
I think generateBoard could be removed from this line, since it is invoked on row 238.

Reply:
No, the board doesn't regenerate with the filtered characters if I remove it.